torch/_dynamo/variables/ctx_manager.py

```diff
@@ -977,6 +977,83 @@ def fn_name(self):
         return "use_training_state"
 
 
+class SDPAKernelVariable(ContextWrappingVariable):
```

Comment:
Do all context managers need Dynamo variables? that seems hard to scale..

I think have 1 source of truth for this lest would be better

Reply:
If we want to support graph breaks in context manager regions, then we need a variable to set context state correctly (most of `SDPAKernelVariable` deals with setting up the correct SDPA kernel state).